### PR TITLE
[Jetpack Full Plugin] Fix JP connected sites not being shown after login on Jetpack

### LINKED ISSUE #18009: [Jetpack Full Plugin] Jetpack connected sites not showing after login (need to be added)
This issue was discovered during the beta testing of Jetpack version 21.8-rc-1.

Ref: pcdRpT-1T8-p2#comment-3286

The feature is supposed to warn the user that a self-hosted site containing a connected Jetpack individual plugin (e.g.: Search, Social) that they need to install the full Jetpack plugin to have proper support inside the app, but on Android the card and dialog are not being shown when we would expect them to be.

Ideally, right after logging in, the self-hosted sites containing connected Jetpack individual plugins should be shown in the site picker list. 

Since they are not, it's currently possible to add them as `self-hosted`, but doing so should cause the dialog to show up and the warning card to appear in the dashboard, but instead, this is what happens:
- After adding the site, nothing shows up, and usually, the dashboard falls into an “infinite loading” state when trying to refresh it.
  - _Note_: infinite loading fixed in #18012
- Restarting the app causes the card to show up after a while, but not the warning fragment / overlay.
- Restarting again (or just going to another screen and back to the dashboard) finally causes the warning fragment / overlay to show up.

### Update (root cause)
We discovered that the real root cause here is that we are not properly fetching the full list of sites right after logging in. So the Jetpack-connected sites are not being listed in the sites list, making it possible to add them as self-hosted, but at later restarts, those sites are fetched but then we have some sort of conflict since the site is both returned by WP.com APIs, and is already added as self-hosted.

The fix will likely involve fetching the correct sites list after login, so connected sites are also shown, which should avoid this problem, as the site will be fetched from the correct .com API.

### Video

https://user-images.githubusercontent.com/5091503/221200801-68dfb9d9-ea78-497e-bfbb-43ad7fbba739.mov

### Expected behavior
Display the dialog and card as soon as the site matching the requirements is added.

### Actual behavior
The expected behavior only happens after restarting the app a couple of times.

### Steps to reproduce the behavior
1. Login to Jetpack with a WP.com
2. Add a self-hosted site that:
  - Doesn't contain the full Jetpack plugin
  - Contains at least one individual Jetpack plugin (e.g.: Search, Social)
  - The Jetpack plugin is connected to the WP.com account from step 1
3. **Verify** the "Install Jetpack Plugin" dialog is not shown
4. **Verify** the card is also not shown in the dashboard
5. Close the app (remove from recents)
6. Open it again
7. **Verify** the card shows after a while in the dashboard
8. **Verify** the dialog is not shown
9. Press home to minimize the app
10. Open it again
11. **Verify** the dialog is finally displayed

### Important
This might not be as major as initially thought since it is more likely users already have their self-hosted sites added when they update to 21.8, which means the pop-up and card **should** work properly.

##### Tested on AS Emulator, Android 13, JPAndroid 21.8-rc-1

### PR DESCRIPTION
**Pull request by RenanLukas**
Fixes #18009 

This PR partially reverts https://github.com/wordpress-mobile/WordPress-Android/pull/17094 so that we can stop filtering JP connected sites on `WordPress-Login-Flow-Android` when it's a Jetpack app login.

Companion PR on `WordPress-Login-Flow-Android` [here](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/116) 

To test:
**Jetpack**
- Log in with an account that has at least one site with at least one individual JP plugin connected (e.g. Jetpack Social) and NOT the full JP plugin;
- Observe the sites list that appears after login: the JP individual plugin connected self-hosted sites should be listed.

**WordPress**
- Log in with an account that has at least one site with at least one individual JP plugin connected (e.g. Jetpack Social) and NOT the full JP plugin;
- Observe the sites list that appears after login: the JP individual plugin connected self-hosted sites should NOT be listed.

## Regression Notes
1. Potential unintended areas of impact
Jetpack login.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Changes were only made to a .gradle file and on the View level (which we're not testing at the moment).

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by zwarm** [comment]
@RenanLukas 
I tested using the instructions you provided. I had mixed results.

✅  Testing in Jetpack works as expected; the self-hosted site is shown in the sites list upon login.
❌ Testing in WP did not work as expected; the self-hosted site is shown in the sites list upon login. 

**Steps I took to test**
- Logged into JP with a self-hosted site that did not have JP (full or any plugins)
- Tapped stats, to get to the install JP prompt, and followed the steps to install plugin (including connecting to WP.com account)
- Was then prompted to install the full JP and waited for the success message
- Logged out of JP app + cleared app data
- Logged into JP and verified site was in sites list
- Installed WP app from this PR
- Logged in WP (same wp.com account from the process above)
- Self hosted connected JP site was shown in the sites list

**Comment by RenanLukas** [comment]
Hi @zwarm , thanks for taking the time to test my PR 👍 

My testing instructions were not clear, sorry about that. The sites that should not be filtered on JP app but should be filtered on WP app are sites that have only individual JP plugins connected (e.g. Jetpack Social) and not the ones with the full JP plugin.

I've updated the testing instructions. Please let me know if they're still not clear. Thanks!

**Review by thomashorta — APPROVED** [review]
LGTM! :shipit: 

Thanks, @RenanLukas for looking int this issue and finding the best way to fix it. I tested various scenarios (including the ones from the PR description), and everything works correctly both in Jetpack and WordPress.

Thanks for the clarification regarding sites with individual plugins and sites with the Jetpack full plugin.

**Review by zwarm — APPROVED** [review]
@RenanLukas 
Thanks for the updated instructions; both test scenario's pass as expected. Thanks for wrangling this. 👍

File: build.gradle [hunk 1]
```diff
@@ -23,7 +23,7 @@ ext {
     gutenbergMobileVersion = 'v1.92.0'
     wordPressAztecVersion = 'v1.6.3'
     wordPressFluxCVersion = 'trunk-fcede535112327abf59416b7b544cf526c44fd00'
-    wordPressLoginVersion = '1.0.0'
+    wordPressLoginVersion = '116-9e75c5927871287fccb0617f15f69bb066656570'
```
Comment (oguzkocer) [inline build.gradle:26]: @RenanLukas Just a reminder that this artifact is published to test a [PR](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/116) and shouldn't be merged to WPAndroid's `trunk`. Our dependencies should always be a tag or an artifact published from `trunk` (such as `trunk-{commit-sha}`) before merging WPAndroid PRs.

I am addressing this as part of the code freeze, but wanted to leave this reminder for future PRs. Let me know if you have any questions.

**Pull request merged**